crates/threshold-bls/src/sig/blind.rs

```diff
@@ -53,7 +53,10 @@ where
     type Error = BlindError;
 
     fn blind_msg<R: RngCore>(msg: &[u8], rng: &mut R) -> (Self::Token, Vec<u8>) {
-        let r = I::Private::rand(rng);
+        let mut r = I::Private::rand(rng);
+        while r == I::Private::zero() || r == I::Private::one() {
+            r = I::Private::rand(rng);
+        }
```

Comment:
Looking at this now, I'd actually recommend panicking if the value is zero or one instead of trying again. It should happen with negligible probability when using a correct RNG, so if we encounter zero or one, we can assume we have a broken RNG. If we have a broken RNG, we should not be doing cryptography and it would be best to bail.

Reply:
Makes sense! Changed.